.github/workflows/build-system-test-react.yml

```diff
@@ -0,0 +1,203 @@
+# Description: this workflow runs build mega-apps that have all the connected components.
+
+name: Build System Test Canary - React
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read # This is required for actions/checkout
+
+on:
+  schedule:
+    - cron: '*/15 * * * *' # Run every 15 minutes
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: ci
+    strategy:
+      fail-fast: false
+      matrix:
+        framework: [react]
+        framework-version: [latest]
+        build-tool: [cra]
+        build-tool-version: [latest]
+        pkg-manager: [npm]
+        language: [ts]
+        node-version: [18]
+        include:
+          - framework: react
+            framework-version: latest
+            build-tool: cra
+            build-tool-version: latest
+            pkg-manager: npm
+            language: ts
+            node-version: 16
+          - framework: react
+            framework-version: 16
+            build-tool: cra
+            build-tool-version: latest
+            pkg-manager: npm
+            language: ts
+            node-version: 18
+          - framework: react
+            framework-version: latest
+            build-tool: cra
+            build-tool-version: latest
+            pkg-manager: npm
+            language: js
+            node-version: 18
+          - framework: react
+            framework-version: latest
+            build-tool: cra
+            build-tool-version: latest
+            pkg-manager: yarn
+            pkg-manager-version: 1.22.19
```

Comment:
Why is `pkg-manager-version` required?

Reply:
Because our plan is to test yarn v1 in cra and yarn v3 in vite. Since we don't have plans to test different npm versions, only adding pkg-manager-version here is enough

Reply:
Got it. Instead of specific version `1.22.19`, is it possible to have a generic version like `1`?

Reply:
Yep. that works https://github.com/0618/amplify-ui/actions/runs/4682282681/jobs/8295935101

Reply:
Done d977d5cf9d2fce932ccdd7eea300bce481b9561f